### ISSUE TITLE
Failed to fetch 22.04 repo

### ISSUE THREAD

**Issue by kschrijver** [microsoft/linux-package-repositories#48]
**Describe the issue**
Unable to perform apt-update to fetch packages.microsoft.com/ubuntu/22.04

**When did the issue occur?**
Since this morning, yesterday everything was fine.

**If applicable, what package did you attempt to install, and from which repo?**
apt-get update, not trying to install a certain package
**Steps to Reproduce**
apt-get update

**Actual Result**
`Hit:1 http://azure.archive.ubuntu.com/ubuntu jammy InRelease
Hit:2 http://azure.archive.ubuntu.com/ubuntu jammy-updates InRelease
Hit:3 http://azure.archive.ubuntu.com/ubuntu jammy-backports InRelease
Hit:4 http://security.ubuntu.com/ubuntu jammy-security InRelease
Get:5 https://packages.microsoft.com/ubuntu/22.04/prod jammy InRelease [3065 B]
Get:6 https://packages.microsoft.com/ubuntu/22.04/prod jammy/main amd64 Packages [52.9 kB]
Get:7 https://packages.microsoft.com/ubuntu/22.04/prod jammy/main all Packages [713 B]
Err:7 https://packages.microsoft.com/ubuntu/22.04/prod jammy/main all Packages
  Hash Sum mismatch
  Hashes of expected file:
   - Filesize:713 [weak]
   - SHA512:23a393800e7efebbc2d225680df6eaeb17580c9d691c646b76f4cadb76cadb91d8adb1c8944cd18ac2f21eed4857a012bc5fdafc28a9374cfcebf35abea817dc
   - SHA256:213e24202ba7cabe92df422d201bce8f8cf6daaf0a471258a27aaa94ae9662be
  Hashes of received file:
   - SHA512:aaf1f56f5346d7d7c21e677a5054ab321442696f17618caa7b0f128644c46acfd021e0dc0679cc3738eca74aed4404ec00a536684ff33f9cc53e2ae4e94f17f3
   - SHA256:341456fda1b9f814bcedf5ee0147a9ee2f95b8cd3d50f4bcec9423d3ce61e6b3
   - Filesize:713 [weak]
  Last modification reported: Wed, 08 Mar 2023 22:57:55 +0000
  Release file created at: Wed, 08 Mar 2023 22:52:10 +0000
Fetched 713 B in 0s (1563 B/s)
Reading package lists... Done
E: Failed to fetch https://packages.microsoft.com/ubuntu/22.04/prod/dists/jammy/main/binary-all/Packages.gz  Hash Sum mismatch
   Hashes of expected file:
    - Filesize:713 [weak]
    - SHA512:23a393800e7efebbc2d225680df6eaeb17580c9d691c646b76f4cadb76cadb91d8adb1c8944cd18ac2f21eed4857a012bc5fdafc28a9374cfcebf35abea817dc
    - SHA256:213e24202ba7cabe92df422d201bce8f8cf6daaf0a471258a27aaa94ae9662be
   Hashes of received file:
    - SHA512:aaf1f56f5346d7d7c21e677a5054ab321442696f17618caa7b0f128644c46acfd021e0dc0679cc3738eca74aed4404ec00a536684ff33f9cc53e2ae4e94f17f3
    - SHA256:341456fda1b9f814bcedf5ee0147a9ee2f95b8cd3d50f4bcec9423d3ce61e6b3
    - Filesize:713 [weak]
   Last modification reported: Wed, 08 Mar 2023 22:57:55 +0000
   Release file created at: Wed, 08 Mar 2023 22:52:10 +0000
E: Some index files failed to download. They have been ignored, or old ones used instead.
`
**Expected Result**
to be able to succesfully update the repo
**Screenshots**
![image](https://user-images.githubusercontent.com/127373616/223918098-90073926-7990-4f4f-93af-12ff4876d852.png)

**Additional context**
Was able to update happily for about 8 hours back. now i m unable to do so.

**Comment by kschrijver** [comment]
Also, cleaning /var/lib/apt/lists > apt-get update > still outputs the same Hash sum mismatch.

**Comment by liveans** [comment]
@daviddavis Can you help with this issue?

**Comment by daviddavis** [comment]
Yea, I can take a look. Can you tell me what you get when you run `dig packages.microsoft.com` or `ping packages.microsoft.com`? It would help me to know what server you're hitting.

**Comment by Ziemowit-LH** [comment]
Seems to have been resolved.

**Comment by i2van** [comment]
Not resolved for me.

```
dig packages.microsoft.com

; <<>> DiG 9.18.1-1ubuntu1.3-Ubuntu <<>> packages.microsoft.com
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 19933
;; flags: qr rd ra; QUERY: 1, ANSWER: 3, AUTHORITY: 0, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 512
;; QUESTION SECTION:
;packages.microsoft.com.                IN      A

;; ANSWER SECTION:
packages.microsoft.com. 2412    IN      CNAME   apt-geofence-parent.trafficmanager.net.
apt-geofence-parent.trafficmanager.net. 212 IN CNAME csd-apt-weu-d-4.westeurope.cloudapp.azure.com.
csd-apt-weu-d-4.westeurope.cloudapp.azure.com. 10 IN A 13.81.215.193

;; Query time: 29 msec
;; SERVER: 8.8.8.8#53(8.8.8.8) (UDP)
;; WHEN: Thu Mar 09 17:43:38 EET 2023
;; MSG SIZE  rcvd: 175
```

**Comment by MichailKaramanos** [comment]
Not resolved for me either

**Comment by MichailKaramanos** [comment]
@daviddavis 

`dig packages.microsoft.com` result 👇🏼  

`; <<>> DiG 9.18.1-1ubuntu1.3-Ubuntu <<>> packages.microsoft.com
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 27235
;; flags: qr rd ra; QUERY: 1, ANSWER: 3, AUTHORITY: 0, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 65494
;; QUESTION SECTION:
;packages.microsoft.com.		IN	A

;; ANSWER SECTION:
packages.microsoft.com.	1737	IN	CNAME	apt-geofence-parent.trafficmanager.net.
apt-geofence-parent.trafficmanager.net.	127 IN CNAME csd-apt-weu-d-3.westeurope.cloudapp.azure.com.
csd-apt-weu-d-3.westeurope.cloudapp.azure.com. 10 IN A 40.114.136.21

;; Query time: 16 msec
;; SERVER: 127.0.0.53#53(127.0.0.53) (UDP)
;; WHEN: Thu Mar 09 15:50:24 UTC 2023
;; MSG SIZE  rcvd: 175
`

**Comment by SneWs** [comment]
And here are mine. Ping never responds ... and dig result below as well
```
marcus@hawk:~$ ping packages.microsoft.com
PING csd-apt-weu-d-3.westeurope.cloudapp.azure.com (40.114.136.21) 56(84) bytes of data.
^C
--- csd-apt-weu-d-3.westeurope.cloudapp.azure.com ping statistics ---
17 packets transmitted, 0 received, 100% packet loss, time 16375ms

marcus@hawk:~$ dig packages.microsoft.com

; <<>> DiG 9.18.1-1ubuntu1.3-Ubuntu <<>> packages.microsoft.com
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 26765
;; flags: qr rd ra; QUERY: 1, ANSWER: 3, AUTHORITY: 0, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 65494
;; QUESTION SECTION:
;packages.microsoft.com.		IN	A

;; ANSWER SECTION:
packages.microsoft.com.	2301	IN	CNAME	apt-geofence-parent.trafficmanager.net.
apt-geofence-parent.trafficmanager.net.	149 IN CNAME csd-apt-weu-d-2.westeurope.cloudapp.azure.com.
csd-apt-weu-d-2.westeurope.cloudapp.azure.com. 10 IN A 104.214.230.139

;; Query time: 24 msec
;; SERVER: 127.0.0.53#53(127.0.0.53) (UDP)
;; WHEN: Thu Mar 09 16:58:57 CET 2023
;; MSG SIZE  rcvd: 175
```

**Comment by daviddavis** [comment]
This problem should be resolved. Apologies. We're looking at how to better to keep the nodes in sync going forward.

**Comment by SneWs** [comment]
Thank you for the quick turn around!

**Comment by kschrijver** [comment]
Seems to be working again! Thank you David!

**Comment by viral1701** [comment]
The issue looks to be unresolved I'm having an issue installing powershell on image 22.04. Is anyone else having issues?

**Comment by i2van** [comment]
> The issue looks to be unresolved I'm having an issue installing powershell on image 22.04. Is anyone else having issues?

Have you tried [this](https://learn.microsoft.com/en-us/powershell/scripting/install/install-ubuntu?view=powershell-7.3)?

**Comment by Peperud** [comment]
Not resolved for me either. 
Verbose output from ```apt``` :
```
0% [Working] -> https:600%20URI%20Acquire%0aURI:%20https://packages.microsoft.com/ubuntu/22.04/prod/pool/main/d/dotnet-sdk-8.0/dotnet-sdk-8.0_8.0.203-1_amd64.deb%0aFilename:%20/var/cache/apt/archives/partial/dotnet-sdk-8.0_8.0.203-1_amd64.deb%0aExpected-SHA256:%2064a74a6cbe6bce4363c8475742334f51d7101f3903977c500e36f96f2ae38a93%0aExpected-Checksum-FileSize:%20110251086%0aTarget-Repo-URI:%20https://packages.microsoft.com/ubuntu/22.04/prod/%0aTarget-Site:%20https://packages.microsoft.com/ubuntu/22.04/prod%0aTarget-Release:%20jammy%0aTarget-Base-URI:%20https://packages.microsoft.com/ubuntu/22.04/prod/%0aTarget-Component:%20main%0aTarget-Codename:%20jammy%0aTarget-Suite:%20jammy%0aTarget-Architecture:%20amd64%0aTarget-Type:%20deb%0a%0a
0% [Working] <- https:102%20Status%0aMessage:%20Connecting%20to%20packages.microsoft.com%20(2620:1ec:bdf::69)%0aURI:%20https://packages.microsoft.com/ubuntu/22.04/prod/pool/main/d/dotnet-sdk-8.0/dotnet-sdk-8.0_8.0.203-1_amd64.deb
 <- https:102%20Status%0aMessage:%20Connecting%20to%20packages.microsoft.com%20(13.107.213.69)%0aURI:%20https://packages.microsoft.com/ubuntu/22.04/prod/pool/main/d/dotnet-sdk-8.0/dotnet-sdk-8.0_8.0.203-1_amd64.deb
 <- https:102%20Status%0aMessage:%20Connected%20to%20packages.microsoft.com%20(13.107.213.69)%0aURI:%20https://packages.microsoft.com/ubuntu/22.04/prod/pool/main/d/dotnet-sdk-8.0/dotnet-sdk-8.0_8.0.203-1_amd64.deb
 <- https:102%20Status%0aMessage:%20Waiting%20for%20headers%0aURI:%20https://packages.microsoft.com/ubuntu/22.04/prod/pool/main/d/dotnet-sdk-8.0/dotnet-sdk-8.0_8.0.203-1_amd64.deb
 <- https:102%20Status%0aMessage:%20Connecting%20to%20packages.microsoft.com%20(13.107.246.69)%0aURI:%20https://packages.microsoft.com/ubuntu/22.04/prod/pool/main/d/dotnet-sdk-8.0/dotnet-sdk-8.0_8.0.203-1_amd64.deb
 <- https:102%20Status%0aMessage:%20Connected%20to%20packages.microsoft.com%20(13.107.246.69)%0aURI:%20https://packages.microsoft.com/ubuntu/22.04/prod/pool/main/d/dotnet-sdk-8.0/dotnet-sdk-8.0_8.0.203-1_amd64.deb
 <- https:102%20Status%0aMessage:%20Waiting%20for%20headers%0aURI:%20https://packages.microsoft.com/ubuntu/22.04/prod/pool/main/d/dotnet-sdk-8.0/dotnet-sdk-8.0_8.0.203-1_amd64.deb
0% [Waiting for headers] <- https:400%20URI%20Failure%0aTransient-Failure:%20true%0aMessage:%20Connection%20failed%20[IP:%2013.107.246.69%20443]%0aURI:%20https://packages.microsoft.com/ubuntu/22.04/prod/pool/main/d/dotnet-sdk-8.0/dotnet-sdk-8.0_8.0.203-1_amd64.deb
Err:1 https://packages.microsoft.com/ubuntu/22.04/prod jammy/main amd64 dotnet-sdk-8.0 amd64 8.0.203-1
  Connection failed [IP: 13.107.246.69 443]
E: Failed to fetch https://packages.microsoft.com/ubuntu/22.04/prod/pool/main/d/dotnet-sdk-8.0/dotnet-sdk-8.0_8.0.203-1_amd64.deb  Connection failed [IP: 13.107.246.69 443]
```

url decoded
```
https:600 URI Acquire
URI: https://packages.microsoft.com/ubuntu/22.04/prod/pool/main/d/dotnet-sdk-8.0/dotnet-sdk-8.0_8.0.203-1_amd64.deb
Filename: /var/cache/apt/archives/partial/dotnet-sdk-8.0_8.0.203-1_amd64.deb
Expected-SHA256: 64a74a6cbe6bce4363c8475742334f51d7101f3903977c500e36f96f2ae38a93
Expected-Checksum-FileSize: 110251086
Target-Repo-URI: https://packages.microsoft.com/ubuntu/22.04/prod/
Target-Site: https://packages.microsoft.com/ubuntu/22.04/prod
Target-Release: jammy
Target-Base-URI: https://packages.microsoft.com/ubuntu/22.04/prod/
Target-Component: main
Target-Codename: jammy
Target-Suite: jammy
Target-Architecture: amd64
Target-Type: deb


https:102 Status
Message: Connecting to packages.microsoft.com (2620:1ec:bdf::69)
URI: https://packages.microsoft.com/ubuntu/22.04/prod/pool/main/d/dotnet-sdk-8.0/dotnet-sdk-8.0_8.0.203-1_amd64.deb
 <- https:102 Status
Message: Connecting to packages.microsoft.com (13.107.213.69)
URI: https://packages.microsoft.com/ubuntu/22.04/prod/pool/main/d/dotnet-sdk-8.0/dotnet-sdk-8.0_8.0.203-1_amd64.deb
 <- https:102 Status
Message: Connected to packages.microsoft.com (13.107.213.69)
URI: https://packages.microsoft.com/ubuntu/22.04/prod/pool/main/d/dotnet-sdk-8.0/dotnet-sdk-8.0_8.0.203-1_amd64.deb
 <- https:102 Status
Message: Waiting for headers
URI: https://packages.microsoft.com/ubuntu/22.04/prod/pool/main/d/dotnet-sdk-8.0/dotnet-sdk-8.0_8.0.203-1_amd64.deb
 <- https:102 Status
Message: Connecting to packages.microsoft.com (13.107.246.69)
URI: https://packages.microsoft.com/ubuntu/22.04/prod/pool/main/d/dotnet-sdk-8.0/dotnet-sdk-8.0_8.0.203-1_amd64.deb
 <- https:102 Status
Message: Connected to packages.microsoft.com (13.107.246.69)
URI: https://packages.microsoft.com/ubuntu/22.04/prod/pool/main/d/dotnet-sdk-8.0/dotnet-sdk-8.0_8.0.203-1_amd64.deb
 <- https:102 Status
Message: Waiting for headers
URI: https://packages.microsoft.com/ubuntu/22.04/prod/pool/main/d/dotnet-sdk-8.0/dotnet-sdk-8.0_8.0.203-1_amd64.deb


https:400 URI Failure
Transient-Failure: true
Message: Connection failed [IP: 13.107.246.69 443]
URI: https://packages.microsoft.com/ubuntu/22.04/prod/pool/main/d/dotnet-sdk-8.0/dotnet-sdk-8.0_8.0.203-1_amd64.deb
Err:1 https://packages.microsoft.com/ubuntu/22.04/prod jammy/main amd64 dotnet-sdk-8.0 amd64 8.0.203-1
  Connection failed [IP: 13.107.246.69 443]
```

curl
```
curl -L -O -C - https://packages.microsoft.com/ubuntu/22.04/prod/pool/main/d/dotnet-sdk-8.0/dotnet-sdk-8.0_8.0.203-1_amd64.deb
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
 45  105M   45 48.0M    0     0  31.1M      0  0:00:03  0:00:01  0:00:02 31.1M
curl: (92) HTTP/2 stream 0 was not closed cleanly: INTERNAL_ERROR (err 2)
```

wget
```
wget  https://packages.microsoft.com/ubuntu/22.04/prod/pool/main/d/dotnet-sdk-8.0/dotnet-sdk-8.0_8.0.203-1_amd64.deb--2024-03-29 13:03:38--  https://packages.microsoft.com/ubuntu/22.04/prod/pool/main/d/dotnet-sdk-8.0/dotnet-sdk-8.0_8.0.203-1_amd64.deb
Resolving packages.microsoft.com (packages.microsoft.com)... 13.107.213.71, 13.107.246.71, 2620:1ec:bdf::69, ...
Connecting to packages.microsoft.com (packages.microsoft.com)|13.107.213.71|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 110251086 (105M) [application/octet-stream]
Saving to: ‘dotnet-sdk-8.0_8.0.203-1_amd64.deb’

dotnet-sdk-8.0_8.0.203-1_amd64.deb    76%[====================================================>                 ]  80.00M  6.96MB/s    in 6.4s    

2024-03-29 13:03:45 (12.5 MB/s) - Connection closed at byte 83886080. Retrying.

--2024-03-29 13:03:46--  (try: 2)  https://packages.microsoft.com/ubuntu/22.04/prod/pool/main/d/dotnet-sdk-8.0/dotnet-sdk-8.0_8.0.203-1_amd64.deb
Connecting to packages.microsoft.com (packages.microsoft.com)|13.107.213.71|:443... connected.
HTTP request sent, awaiting response... 206 Partial Content
Length: 110251086 (105M), 26365006 (25M) remaining [application/octet-stream]
Saving to: ‘dotnet-sdk-8.0_8.0.203-1_amd64.deb’

dotnet-sdk-8.0_8.0.203-1_amd64.deb    83%[+++++++++++++++++++++++++++++++++++++++++++++++++++++====>            ]  88.00M  22.3MB/s    in 0.4s    

2024-03-29 13:03:46 (22.3 MB/s) - Connection closed at byte 92274688. Retrying.

--2024-03-29 13:03:48--  (try: 3)  https://packages.microsoft.com/ubuntu/22.04/prod/pool/main/d/dotnet-sdk-8.0/dotnet-sdk-8.0_8.0.203-1_amd64.deb
Connecting to packages.microsoft.com (packages.microsoft.com)|13.107.213.71|:443... connected.
HTTP request sent, awaiting response... 206 Partial Content
Length: 110251086 (105M), 17976398 (17M) remaining [application/octet-stream]
Saving to: ‘dotnet-sdk-8.0_8.0.203-1_amd64.deb’

dotnet-sdk-8.0_8.0.203-1_amd64.deb   100%[++++++++++++++++++++++++++++++++++++++++++++++++++++++++++===========>] 105.14M  21.7MB/s    in 0.8s    

2024-03-29 13:03:49 (21.7 MB/s) - ‘dotnet-sdk-8.0_8.0.203-1_amd64.deb’ saved [110251086/110251086]
```

**Comment by daviddavis** [comment]
@Peperud your issue is totally unrelated to the one here. Can you open a new issue? Also, having the headers for your curl request would help us to debug. You can use `-vvv` or `--head`.
